src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1357,6 +1357,7 @@ def prepare_stencil_definition(cls, definition, externals):
             definition._gtscript_["nonlocals"], definition._gtscript_["imported"], externals
         )
         definition._gtscript_["externals"] = resolved_externals
+        return definition
```

Comment:
This is used by the builder to provide the definition always in annotated form, because some other parts of the code rely on `_gtscript_` being present without checking. The tests were failing at some point due to that.

Another consecuence of that was the addition of the `AnnotatedFunctionDef` protocol, so long term we can annotate where exactly this is required.